### PR TITLE
feat: Socket.io implementation in extension

### PR DESCRIPTION
**Pull request by mayank-SFIN571**
This PR adds implementation for socket.io as a means to send real time messages from server to clients.

Test cases are pending for this.

### CODE REVIEW

File: src/providers/notification.provider.ts [hunk 1]
```diff
@@ -21,6 +23,8 @@ export class NotificationProvider implements Provider<INotification> {
       return this.emailProvider.publish(data);
     } else if (data.type === MessageType.Push && this.pushProvider) {
       return this.pushProvider.publish(data);
+    } else if (data.type === MessageType.SocketIO && this.socketProvider) {
```
Comment (samarpanB) [inline src/providers/notification.provider.ts:26]: socket io is not a new message type. Mesage type will remain push. socket io is just a new push provider.

**Review by samarpanB — CHANGES_REQUESTED** [review]

File: src/types.ts [hunk 2]
```diff
@@ -46,4 +46,5 @@ export const enum MessageType {
   Push,
   Email,
   SMS,
+  SocketIO,
```
Comment (samarpan-b) [inline src/types.ts:49]: remove tthis

**Review by samarpan-b — COMMENTED** [review]
Add docs to readme with usage details

File: README.md [hunk 3]
```diff
@@ -279,6 +280,64 @@ If you wish to use any other service provider of your choice, you can create a p
...
+    ....
+
+    this.component(NotificationsComponent);
+    this.bind(NotificationBindings.PushProvider).toProvider(SocketIOProvider);
+    ....
+  }
+}
+```
+
+There are some additional configurations needed in order to allow Pubnub connection. You need to add them as below. Make sure these are added before the provider binding.
```
Comment (samarpan-b) [inline README.md:308]: why pubnub ?

File: package.json [hunk 4]
```diff
@@ -45,6 +45,8 @@
     "@loopback/context": "^3.8.1",
     "@loopback/core": "^2.7.0",
     "@loopback/rest": "^5.0.1",
+    "@types/socket.io-client": "^1.4.33",
```
Comment (samarpan-b) [inline package.json:48]: add to dev deps

**Review by samarpan-b — CHANGES_REQUESTED** [review]

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**